bundles/event_admin/event_admin/src/celix_event_admin.c

```diff
@@ -415,6 +454,131 @@ int celix_eventAdmin_removeEventHandlerWithProperties(void* handle, void* svc, c
     return CELIX_SUCCESS;
 }
 
+int celix_eventAdmin_addRemoteProviderService(void* handle, void* svc, const celix_properties_t* props) {
+    assert(handle != NULL);
+    assert(svc != NULL);
+    celix_event_admin_t* ea = handle;
+    long serviceId = celix_properties_getAsLong(props, CELIX_FRAMEWORK_SERVICE_ID, -1L);
+    if (serviceId < 0) {
```

Comment:
IMO this should not be possible, so maybe a `assert(serviceId >= 0);` is enough; this saves some test code.